### PR TITLE
script/sync:add support to sync nodes

### PR DESCRIPTION
**Pull request by blackjid**
Adds support to sync nodes to S3

The PR https://github.com/boxen/puppet-nodejs/pull/48 make use of this.

### CODE REVIEW

**Comment by blackjid** [comment]
The node binaries are only synced when no arguments are passed to the `sync` command.

I thought that may be this could be something like this.

```shell
script/sync #sync everything
script/sync --brew emacs/24.3-boxen2
script/sync --ruby 1.9.3-p448
script/sync --node 0.12.2
``` 

Any other idea?

File: script/sync [hunk 1]
```diff
@@ -134,6 +134,30 @@ def sync_ruby(version)
   puts "done."
 end
 
+def sync_node(version)
+  s3_key = "nodes/Darwin/#{s3_cellar}#{os}/#{version}.tar.bz2"
```
Comment (mikemcquaid) [inline script/sync]: Do these link to anything in Boxen's Homebrew directory?

Comment (blackjid) [inline script/sync]: Hi, mmm. I'm not really sure... I took the idea from the ruby sync function.
In the rewrite I proposed, the provider is used in the nodebuild provider.
https://github.com/platanus/puppet-nodejs/blob/rewrite/lib/puppet/provider/nodejs/nodebuild.rb#L75

Do you think there is something wrong??

Comment (mikemcquaid) [inline script/sync]: Can you check files with `otool -L` to see if there's any linkage to the `/opt/boxen/homebrew` or `/usr/local` directories? Thanks!

Comment (blackjid) [inline script/sync]: He mike, sorry... but I don't really understand what I'm doing nor how to do it.... can you expand of what are we checking?? thanks! 😁

Comment (mikemcquaid) [inline script/sync]: Use `otool -L` on the files inside `/opt/nodes/`  and see if any of them reference any paths in `/opt/boxen/homebrew`

Comment (blackjid) [inline script/sync]: Thanks!, I doesn't seems to be any references..
I did run this
```
find /opt/nodes/. -type f -print | xargs otool -L
```

This was the output
```
/opt/nodes/./0.10.36/bin/node:
	/System/Library/Frameworks/Carbon.framework/Versions/A/Carbon (compatibility version 2.0.0, current version 157.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1213.0.0)
	/System/Library/Frameworks/Foundation.framework/Versions/C/Foundation (compatibility version 300.0.0, current version 1153.20.0)
	/System/Library/Frameworks/CoreServices.framework/Versions/A/CoreServices (compatibility version 1.0.0, current version 62.0.0)
	/System/Library/Frameworks/ApplicationServices.framework/Versions/A/ApplicationServices (compatibility version 1.0.0, current version 48.0.0)
	/usr/lib/libstdc++.6.dylib (compatibility version 7.0.0, current version 104.1.0)
	/usr/lib/libgcc_s.1.dylib (compatibility version 1.0.0, current version 283.0.0)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1153.18.0)
/opt/nodes/./0.10.36/include/node/ares.h: is not an object file
/opt/nodes/./0.10.36/include/node/ares_version.h: is not an object file
/opt/nodes/./0.10.36/include/node/common.gypi: is not an object file
/opt/nodes/./0.10.36/include/node/config.gypi: is not an object file
/opt/nodes/./0.10.36/include/node/nameser.h: is not an object file
/opt/nodes/./0.10.36/include/node/node.h: is not an object file
/opt/nodes/./0.10.36/include/node/node_buffer.h: is not an object file
/opt/nodes/./0.10.36/include/node/node_internals.h: is not an object file
/opt/nodes/./0.10.36/include/node/node_object_wrap.h: is not an object file
/opt/nodes/./0.10.36/include/node/node_version.h: is not an object file
/opt/nodes/./0.10.36/include/node/openssl/opensslconf.h: is not an object file
/opt/nodes/./0.10.36/include/node/uv-private/ngx-queue.h: is not an object file
/opt/nodes/./0.10.36/include/node/uv-private/stdint-msvc2008.h: is not an object file
/opt/nodes/./0.10.36/include/node/uv-private/tree.h: is not an object file
/opt/nodes/./0.10.36/include/node/uv-private/uv-bsd.h: is not an object file
/opt/nodes/./0.10.36/include/node/uv-private/uv-darwin.h: is not an object file
/opt/nodes/./0.10.36/include/node/uv-private/uv-linux.h: is not an object file
/opt/nodes/./0.10.36/include/node/uv-private/uv-sunos.h: is not an object file
/opt/nodes/./0.10.36/include/node/uv-private/uv-unix.h: is not an object file
/opt/nodes/./0.10.36/include/node/uv-private/uv-win.h: is not an object file
/opt/nodes/./0.10.36/include/node/uv.h: is not an object file
/opt/nodes/./0.10.36/include/node/v8-debug.h: is not an object file
/opt/nodes/./0.10.36/include/node/v8-preparser.h: is not an object file
/opt/nodes/./0.10.36/include/node/v8-profiler.h: is not an object file
/opt/nodes/./0.10.36/include/node/v8-testing.h: is not an object file
/opt/nodes/./0.10.36/include/node/v8.h: is not an object file
/opt/nodes/./0.10.36/include/node/v8stdint.h: is not an object file
/opt/nodes/./0.10.36/include/node/zconf.h: is not an object file
/opt/nodes/./0.10.36/include/node/zlib.h: is not an object file
/opt/nodes/./0.10.36/lib/dtrace/node.d: is not an object file
/opt/nodes/./0.10.36/share/man/man1/node.1: is not an object file
```

**Comment by mikemcquaid** [comment]
Great, thanks!

**Pull request merged**